src/health/health.controller.ts

```diff
@@ -0,0 +1,45 @@
+import { Controller, Get } from "@nestjs/common";
+import { ApiTags } from "@nestjs/swagger";
+import {
+  DiskHealthIndicator,
+  HealthCheck,
+  HealthCheckService,
+  HttpHealthIndicator,
+  MemoryHealthIndicator,
+} from "@nestjs/terminus";
+import { PrismaHealthIndicatorService } from "@wiredcraft/dbaccess/prisma.health.indicator.service";
+
+@Controller("health")
+@ApiTags("System Health Status")
+export class HealthController {
+  constructor(
+    private readonly http: HttpHealthIndicator,
+    private readonly health: HealthCheckService,
+    private readonly disk: DiskHealthIndicator,
+    private readonly memory: MemoryHealthIndicator,
+    private readonly dbHealth: PrismaHealthIndicatorService
+  ) {}
+
+  @Get()
+  @HealthCheck()
+  check() {
+    return this.health.check([
+      () =>
+        this.http.pingCheck(
+          "basic check",
+          `http://localhost:${process.env.PORT || 3000}`
+        ),
+      () => this.dbHealth.isHealthy("mongodb"),
+      () =>
+        this.disk.checkStorage("diskStorage", {
```

Comment:
why monitor disk storage in the backend service?
does it store anything on disk to support certain feature?

Reply:
not really, just my coding routine. actually I just meant to monitor the database status.